src/Backend/Exasol/ToFinalTable/SqlBuilder.php

```diff
@@ -0,0 +1,385 @@
+    ): string {
+        $dest = sprintf(
+            '%s.%s',
+            ExasolQuote::quoteSingleIdentifier($destinationDefinition->getSchemaName()),
+            ExasolQuote::quoteSingleIdentifier($destinationDefinition->getTableName())
+        );
+
+        $columnsSet = [];
+        /** @var ExasolColumn $columnDefinition */
+        foreach ($stagingTableDefinition->getColumnsDefinitions() as $columnDefinition) {
+            if (in_array($columnDefinition->getColumnName(), $destinationDefinition->getPrimaryKeysNames(), true)) {
+                // primary keys are not updated
+                continue;
+            }
+            $columnsSet[] = sprintf(
+                '%s = "src".%s',
+                ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName()),
+                ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName())
+            );
+        }
+
+        if ($importOptions->useTimestamp()) {
+            $columnsSet[] = sprintf(
+                '%s = \'%s\'',
+                ExasolQuote::quoteSingleIdentifier(ToStageImporterInterface::TIMESTAMP_COLUMN_NAME),
+                $timestamp
+            );
+        }
+
+        // update only changed rows - mysql TIMESTAMP ON UPDATE behaviour simulation
+        $columnsComparisionSql = array_map(
+            function (ExasolColumn $columnDefinition) {
+                return sprintf(
+                // phpcs:ignore
+                    '("dest".%s != "src".%s OR ("dest".%s IS NULL OR "src".%s IS NULL AND ("dest".%s IS NULL AND "src".%s IS NULL)))',
+                    ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName()),
+                    ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName()),
+                    ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName()),
+                    ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName()),
+                    ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName()),
+                    ExasolQuote::quoteSingleIdentifier($columnDefinition->getColumnName())
+                );
+            },
+            iterator_to_array($stagingTableDefinition->getColumnsDefinitions())
+        );
+
+        return sprintf(
+            'UPDATE %s AS "dest" SET %s FROM %s.%s AS "src",%s AS "dest" WHERE %s AND (%s) ',
+            $dest,
+            implode(', ', $columnsSet),
+            ExasolQuote::quoteSingleIdentifier($stagingTableDefinition->getSchemaName()),
+            ExasolQuote::quoteSingleIdentifier($stagingTableDefinition->getTableName()),
+            $dest,
+            $this->getPrimaryKeyWhereConditionsNull(
+                $destinationDefinition->getPrimaryKeysNames(),
+                '"src"',
+                '"dest"'
+            ),
+            implode(' OR ', $columnsComparisionSql)
+        );
+    }
+
+    private function getColumnTypeSqlDefinition(ExasolColumn $column): string
```

Comment:
tohle bych rad nejak vyresil pres php-datatypes, ale minimalne to fixuju tady https://github.com/keboola/php-db-import-export/pull/66/files#diff-a5534a09128dcf3a3274116b2d0c851f12414df365cd881ccdbf0c24e1b11798R379